### PR TITLE
#85 Provide and use ParseResult factory methods.

### PR DESCRIPTION
**Pull request by rdvdijk**
@jvdb What do you think?

(I branched the `dynamic-callbacks` branch here, by the way.)

### CODE REVIEW

File: core/src/main/java/io/parsingdata/metal/token/Seq.java [hunk 1]
```diff
@@ -39,15 +41,22 @@ public Seq(final String name, final Encoding enc, final Token... tokens) {
         return tokens.clone();
     }
 
+    /**
```
Comment (rdvdijk) [inline core/src/main/java/io/parsingdata/metal/token/Seq.java]: Oops!

**Comment by jvdb** [comment]
Looks great! My only nitpick that `success()` is a noun and `fail()` a verb. So it should probably either be `success()` and `error()` or `failure()` (nouns) or `succeed()` and `fail()` (verbs).

But I see how you came to the current ones, they're (by themselves) the best options...

**Comment by rdvdijk** [comment]
Verb and noun, well I'll be damned. Totally missed that. How about we go with `success()` and `failure()`, just because they're both 7 characters long? 😉

**Comment by jvdb** [comment]
Agreed, those are probably the best same-type words :)

**Comment by rdvdijk** [comment]
@jvdb Renamed.

**Pull request merged**